master/buildbot/process/slavebuilder.py

```diff
@@ -191,6 +191,7 @@ def includeSlavename(self, msg):
 
     def pingTimeout(self, value, timeout):
         log.msg("ping %s timeout after %s secs" % (self.slavename, timeout))
+        raise Exception("Ping has timeout")
```

Comment:
I think the reason the `errback` is not being called is because we added the `onTimeoutCancel`:

```rd.addTimeout(timeout, reactor, onTimeoutCancel=self.pingTimeout)```

Without it:

```rd.addTimeout(timeout, reactor)```

It will call the `errback`. So we should probably just get rid of `pingTimeout` all together and move the logging to the `errback`.